server/src/com/cloud/network/router/CommandSetupHelper.java

```diff
@@ -848,13 +849,37 @@ public int compare(final PublicIpAddress o1, final PublicIpAddress o2) {
                 associatedWithNetworkId = ipAddrList.get(0).getNetworkId();
             }
 
+            // for network if the ips does not have any rules, then only last ip
+            List<IPAddressVO> userIps = _ipAddressDao.listByAssociatedNetwork(associatedWithNetworkId, null);
+
+            int ipsWithrules = 0;
+            int ipsStaticNat = 0;
+            for (IPAddressVO ip : userIps) {
+                if ( _rulesDao.countRulesByIpIdAndState(ip.getId(), FirewallRule.State.Active) > 0 ) {
+                    ipsWithrules++;
+                }
+
+                // check onetoonenat and also check if the ip "add":false. If there are 2 PF remove 1 static nat add
+                if (ip.isOneToOneNat() && ip.getRuleState() == null) {
```

Comment:
Similar to the comments about checking if `lastIp` isn't null, I find that checking rule state to be null is a bit ... obscure. What is null supposed to mean in this context? It seems the `rule_state` column is only changed to `Releasing` when releasing an IP and nothing else. I guess that's intentional?